crates/ruma-common/src/directory.rs

```diff
@@ -223,6 +228,29 @@ impl<'de> Visitor<'de> for RoomNetworkVisitor {
     }
 }
 
+/// The rule used for users wishing to join a public room.
+///
+/// This type can hold an arbitrary string. To check for formats that are not available as a
```

Comment:
```suggestion
/// This type can hold an arbitrary string. To check for join rules that are not available as a
```

Reply:
Actually I removed the `_Custom` variant. Should I put it back or remove this paragraph?

Reply:
Why did you remove it?

Reply:
I didn't realize that any value could be forwarded from the room state.

Reply:
In general when an enum like this is not internal (i.e. it implements `Deserialize`) and corresponds to some JSON string used in client-server or server-server communication, it should allow custom values instead of failing to deserialize on unknown input (for forwards compatibility).